cl/corpus_importer/management/commands/make_aws_manifest_from_search.py

```diff
@@ -0,0 +1,186 @@
+        parser.add_argument(
+            "--bucket-name",
+            help="The name of the bucket to store documents.",
+            required=True,
+        )
+        parser.add_argument(
+            "--search-text",
+            type=str,
+            required=True,
+            help="Query string used to filter the data records to be retrieved",
+        )
+        parser.add_argument(
+            "--search-type",
+            type=str,
+            required=True,
+            choices=[
+                SEARCH_TYPES.OPINION,
+            ],
+            help=f"The search type to import: ({', '.join([SEARCH_TYPES.OPINION])})",
+        )
+        parser.add_argument(
+            "--filed-after",
+            default="",
+            type=lambda s: valid_date(s) if s else "",
+            help="""Filter documents filed **on or after** the specified date.
+            Format: MM/DD/YYYY (e.g., 12/31/2023).""",
+        )
+        parser.add_argument(
+            "--filed-before",
+            default="",
+            type=lambda s: valid_date(s) if s else "",
+            help="""Filter documents filed **on or before** the specified date.
+            Format: MM/DD/YYYY (e.g., 12/31/2023).""",
+        )
+        parser.add_argument(
+            "--es-page-size",
+            type=int,
+            default=100,
+            help="The number of records to query in a single ES page.",
+        )
+
+    def handle(self, *args, **options):
+        bucket_name = options["bucket_name"]
+        record_type = options["search_type"]
+        search_query = build_base_query(options)
+        has_next = True
+        search_after = None
+        with io.StringIO() as csvfile:
+            writer = csv.DictWriter(
+                csvfile,
+                fieldnames=["bucket", "file_name"],
+                extrasaction="ignore",
+            )
+            while has_next:
+                if search_after:
+                    search_query = search_query.extra(
+                        search_after=search_after
+                    )
+                results = search_query.execute()
+                # This step ensures we only process the exact number of
+                # documents per page defined when calling the command,
+                # effectively removing the extra document originally included
+                # to check for the existence of a next page.
+                clean_page_data = results.hits[: options["es_page_size"]]
+
+                # Create an underscore-separated file name that lambda
+                # can split and use as part of batch processing.
+                ids = [str(r.id) for r in clean_page_data]
+                query_dict = {
+                    "bucket": bucket_name,
+                    "file_name": "_".join(ids),
+                }
+                writer.writerow(query_dict)
+
+                has_next = len(results) > options["es_page_size"]
+                search_after = clean_page_data[-1].meta.sort
```

Comment:
In case a query doesn't match any results, this can throw an index error. I think we can do something like:

```suggestion
                if len(results) > 0:
                    search_after = clean_page_data[-1].meta.sort
```

Reply:
> this can throw an index error.

You're right, Great catch! I'm glad you noticed 👍